### DISCUSSION TITLE
How to embed PDF viewer

### DISCUSSION THREAD

**Discussion by apandiani** [zauberzeug/nicegui#1132] [Q&A] [answered]
### Question

I have tried to embed a PDF viewer, but all I see is a blank area and no error. Can anyone help with this. It should be a native app accessing local PDF files.

Thank you!

```python
@ui.refreshable
def pdf_viewer():
    path = f'file:///{d.full_path}'
    ui.html(f'<embed src="{path}" type="application/pdf" width="100%" height="100%">')
```

**Comment by falkoschindler** [comment, marked as answer]
JavaScript says "Not allowed to load local resource". So you can't embed a "file:///..." resource directly.
But you can add the PDF as a static file and embed the automatically generated `path`:
```py
path = app.add_static_file(local_file='/Users/falko/some.pdf')
ui.html(f'<embed src="{path}" type="application/pdf" width="100%" height="100%">')
```

**Reply by apandiani** [reply]
Thanks a lot!

**Reply by jacoverster** [reply]
This is great, is it possible to select/highlight a word in the PDF and access that value in the nicegui app somehow?
